sklearn/preprocessing/tests/test_data.py

```diff
@@ -1651,3 +1654,50 @@ def test_fit_cold_start():
         # with a different shape, this may break the scaler unless the internal
         # state is reset
         scaler.fit_transform(X_2d)
+
+
+def test_boxcox_transform():
+    # Apply boxcox transform and check if it is
+    # applying individually on each feature
+    X = np.array([[4, 2, 1], [1, 6, 3], [1, 5, 2], [3, 1, 3]])
+    Xtr = boxcox(X)
+    n_features = X.shape[1]
+    for k in range(n_features):
+        assert_array_equal(stats.boxcox(X[:, k])[0], Xtr[:, k])
+        assert_array_equal(boxcox(X[:, k:k + 1]).ravel(),
+                           Xtr[:, k])
+
+
+def test_boxcox_transformer():
+    rng = np.random.RandomState(42)
+    X = rng.randn(3000, 2)
+    lambda0 = 0.1
+    X[:, 0] = np.exp(X[:, 0])
+    X[:, 1] = ((X[:, 1] * lambda0 + 1)) ** (1. / lambda0)
+
+    feature_indices = [0]
+    bct = BoxCoxTransformer(feature_indices=feature_indices)
+
+    bct.fit(X)
+    assert_true(len(bct.lambdas_), 1)
+    X_tr = bct.transform(X)
+    assert_true(X_tr.shape, X.shape)
+    assert_true(np.min(X_tr[:, 0]) < 0.)
```

Comment:
why don't you just keep the original randn data and `assert_allclose(X_tr[:, 0], X_orig[:, 0])`?